typescript/agentkit/src/action-providers/jupiter/jupiterActionProvider.ts

```diff
@@ -0,0 +1,107 @@
+import { ActionProvider } from "../actionProvider";
+import { Network } from "../../network";
+import { SvmWalletProvider } from "../../wallet-providers/svmWalletProvider";
+import { z } from "zod";
+import { CreateAction } from "../actionDecorator";
+import { SwapTokenSchema } from "./schemas";
+import { PublicKey, VersionedTransaction } from "@solana/web3.js";
+import { createJupiterApiClient, SwapRequest } from "@jup-ag/api";
+
+/**
+ * JupiterActionProvider handles token swaps using Jupiter's API.
+ */
+export class JupiterActionProvider extends ActionProvider<SvmWalletProvider> {
+  /**
+   * Initializes Jupiter API client.
+   */
+  constructor() {
+    super("jupiter", []);
+  }
+
+  /**
+   * Swaps tokens using Jupiter's API.
+   *
+   * @param walletProvider - The wallet provider to use for the swap
+   * @param args - Swap parameters including input token, output token, and amount
+   * @returns A message indicating success or failure with transaction details
+   */
+  @CreateAction({
+    name: "swap",
+    description: `
+    Swaps tokens using Jupiter's DEX aggregator.
+    - Input and output tokens must be valid SPL token mints.
+    - Ensures sufficient balance before executing swap.
```

Comment:
Where does the balance check occur? Is it part of the Jup API?

Reply:
Yes